libcudacxx/test/libcudacxx/std/containers/views/views.span/span.cons/initializer_list.pass.cpp

```diff
@@ -1,50 +1,129 @@
 //===----------------------------------------------------------------------===//
 //
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 // SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES.
 //
 //===----------------------------------------------------------------------===//
-// UNSUPPORTED: c++03, c++11
 
-// <span>
+// UNSUPPORTED: c++11
 
+// <cuda/std/span>
+
+// constexpr explicit(extent != dynamic_extent) span(std::initializer_list<value_type> il);
+
+// #include <any>
 #include <cuda/std/cassert>
 #include <cuda/std/cstddef>
+#include <cuda/std/initializer_list>
 #include <cuda/std/span>
+#include <cuda/std/type_traits>
 
+#include "test_convertible.h"
 #include "test_macros.h"
+
+// SFINAE
+
+template <typename T>
+struct ConstElementType : cuda::std::is_const<typename T::element_type>
+{};
+
+static_assert(ConstElementType<cuda::std::span<const int>>::value, "");
+static_assert(!ConstElementType<cuda::std::span<int>>::value, "");
+static_assert(ConstElementType<cuda::std::span<const int, 94>>::value, "");
+static_assert(!ConstElementType<cuda::std::span<int, 94>>::value, "");
+
+// Constructor constraings
+
+template <typename I, typename T, cuda::std::size_t N = cuda::std::dynamic_extent, typename = void>
+struct HasInitializerListCtr : cuda::std::false_type
+{};
+
+template <typename I, typename T, cuda::std::size_t N>
+struct HasInitializerListCtr<I, T, N, cuda::std::void_t<decltype(cuda::std::span<T, N>{I{}})>> : cuda::std::true_type
+{};
+
+static_assert(HasInitializerListCtr<cuda::std::initializer_list<const int>, const int>::value, "");
+static_assert(!HasInitializerListCtr<cuda::std::initializer_list<int>, int>::value, "");
+static_assert(HasInitializerListCtr<cuda::std::initializer_list<const int>, const int, 94>::value, "");
+static_assert(!HasInitializerListCtr<cuda::std::initializer_list<int>, int, 94>::value, "");
+
+// Constructor conditionally explicit
+
+static_assert(!test_convertible<cuda::std::span<const int, 28>, cuda::std::initializer_list<int>>(),
+              "This constructor must be explicit");
+static_assert(cuda::std::is_constructible<cuda::std::span<const int, 28>, cuda::std::initializer_list<int>>::value, "");
+static_assert(test_convertible<cuda::std::span<const int>, cuda::std::initializer_list<int>>(),
+              "This constructor must not be explicit");
+static_assert(cuda::std::is_constructible<cuda::std::span<const int>, cuda::std::initializer_list<int>>::value, "");
+
 struct Sink
 {
   constexpr Sink() = default;
-  __host__ __device__ constexpr Sink(Sink*) {}
+  constexpr __host__ __device__ Sink(Sink*) {}
```

Comment:
We always has `__host__ __device__` and other attributes at the front
```suggestion
  __host__ __device__ constexpr Sink(Sink*) {}
```